tools/src/main/java/org/apache/kafka/tools/ProducerPerformance.java

```diff
@@ -130,42 +142,178 @@ private static ArgumentParser argParser() {
+            this.selectRateSamples = new ArrayList<Double>();
+            this.requestRateSamples = new ArrayList<Double>();
+            this.producer = producer;
+            this.clientId = clientId;
+            this.tags = new HashMap<String, String>();
+            this.tags.put("client-id", clientId);
+        }
+
+        public void recordMetricStats() {
+            Map<MetricName, Metric> metrics = producer.metrics();
+            selectRateSamples.add(metrics.get(new MetricName("select-rate", "producer-metrics", "", tags)).value());
+            requestRateSamples.add(metrics.get(new MetricName("request-rate", "producer-metrics", "", tags)).value());
+        }
+
+        public void printMetricStats() {
+            recordMetricStats();
+            Map<MetricName, Metric> metrics = producer.metrics();
+            double selectRateAvg = avg(selectRateSamples);
+            double requestRateAvg = avg(requestRateSamples);
+            double batchSizeAvg = metrics.get(new MetricName("batch-size-avg", "producer-metrics", "", tags)).value();
+            double requestSizeAvg = metrics.get(new MetricName("request-size-avg", "producer-metrics", "", tags)).value();
+            double recordQueueTimeAvg = metrics.get(new MetricName("record-queue-time-avg", "producer-metrics", "", tags)).value();
+            double requestLatencyAvg = metrics.get(new MetricName("request-latency-avg", "producer-metrics", "", tags)).value();
+            double recordsPerRequestAvg = metrics.get(new MetricName("records-per-request-avg", "producer-metrics", "", tags)).value();
+            double compressionRateAvg = metrics.get(new MetricName("compression-rate-avg", "producer-metrics", "", tags)).value();
+
+            System.out.println(String.format("Select_Rate_Avg:          %.2f", selectRateAvg));
+            System.out.println(String.format("Request_Rate_Avg:         %.2f", requestRateAvg));
+            System.out.println(String.format("Request_Latency_Avg:      %.2f", requestLatencyAvg));
+            System.out.println(String.format("Request_Size_Avg:         %.2f", requestSizeAvg));
+            System.out.println(String.format("Batch_Size_Avg:           %.2f", batchSizeAvg));
+            System.out.println(String.format("Records_Per_Request_Avg:  %.2f", recordsPerRequestAvg));
+            System.out.println(String.format("Record_Queue_Time_Avg:    %.2f", recordQueueTimeAvg));
+            System.out.println(String.format("Compression_Rate_Avg:     %.2f", compressionRateAvg));
+        }
+
+        private double avg(List<Double> samples) {
+            double sum = 0.0;
+            for (Double value : samples)
+                sum += value;
+            return samples.size() == 0 ? -1.0 : sum / samples.size();
+        }
+    }
+
     private static class Stats {
+        private ThreadStats[] threadStatsArr;
+        private long start;
+
+        public Stats(long numRecords, int reportingInterval, int numThreads) {
```

Comment:
Do you want to make this `reportingInterval` configurable?

Since it represents time in miliseconds, do you want to name it as `reportingIntervalMs` and use `long` instead of `int` as its type?